osbs/repo_utils.py

```diff
@@ -60,10 +60,30 @@ def __init__(self, dir_path='', file_name=REPO_CONFIG_FILE):
             with open(file_path) as f:
                 self.container = (yaml.load(f) or {})
 
+        modules = self.container.get('compose', {}).get('modules', [])
+        self.container_module_specs = [ModuleSpec.from_text(module) for module in modules]
+
     def is_autorebuild_enabled(self):
         return self._config_parser.getboolean('autorebuild', 'enabled')
 
 
+class ModuleSpec(object):
+    __slots__ = ('name', 'stream', 'version')
+
+    def __init__(self, name, stream, version=None):
+        self.name = name
+        self.stream = stream
+        self.version = version
+
+    @classmethod
+    def from_text(cls, text):
+        pieces = text.split(':')
+        if not 1 < len(pieces) < 4:
```

Comment:
Now we're instantiating the class this check is done for us. Could either leave it to propagate TypeError, or we could catch TypeError and re-raise as ValueError with the message as we do now.

Reply:
Should we catch attribute error on the line above (split) and re-raise as ValueError instead?

Reply:
So regexp would be good probably for this

Reply:
In atomic-reactor, @owtaylor  added a few comments about not allowing partial entries (which would allow some empty fields). Would this apply here or is it only valid when the CONTEXT field is used?

Reply:
OK, I am checking for empty strings so we will not be more restrictive than ODCS itself